e2e/README.md

```diff
@@ -0,0 +1,92 @@
+# Longhorn e2e tests
+
+### Requirement
+
+1. A Kubernetes cluster with 3 worker nodes.
+   - And control node(s) with following taints:
+      - `node-role.kubernetes.io/master=true:NoExecute`
+      - `node-role.kubernetes.io/master=true:NoSchedule` 
+2. Longhorn system has already been successfully deployed in the cluster.
+3. Make sure MountPropagation feature is enabled
+   - For RKE before v0.1.9, you would need the extra parameter feature-gates: `MountPropagation=true` for kube-api and kubelet to enable the feature.
+4. Make sure `nfs-common` or equivalent has been installed on the node to allow the NFS client to work.
+
+### Run the test
+
+1. Deploy all backupstore servers (including `NFS` server and `Minio` as s3 server) for test purposes.
+```
+kubectl create -Rf integration/deploy/backupstores
+```
+
+2. expose Longhorn API:
+```
+# for example, using nodeport:
+kubectl expose --type=NodePort deployment longhorn-ui -n longhorn-system --port 8000 --name longhorn-ui-nodeport --overrides '{ "apiVersion": "v1","spec":{"ports": [{"port":8000,"protocol":"TCP","targetPort":8000,"nodePort":30000}]}}'
+# or using port-forward:
+kubectl port-forward services/longhorn-frontend 8080:http -n longhorn-system
+```
+
+3. export environment variable `KUBECONFIG`:
+```
+export KUBECONFIG=/path/to/your/kubeconfig.yaml
+```
+
+4. export environment variable `LONGHORN_CLIENT_URL`:
+```
+# for example, if it's exposed by nodeport:
+export LONGHORN_CLIENT_URL=http://node-public-ip:30000
+# or exposed by port-foraword:
+export LONGHORN_CLIENT_URL=http://localhost:8080
+```
+
+5. Prepare test environment and run the test
+```
+cd e2e
+python -m venv .
+source bin/activate
+pip install -r requirements.txt
```

Comment:
Suggest using a dockerfile, so users can just run docker instead. This can be environment independent.

Reply:
Need to figure out how to access k8s cluster in the container, since KUBECONFIG file is in local and not visible by the container.

Created https://github.com/longhorn/longhorn/issues/6824 for this.

Reply:
Probably can just volume binding and host privilege to achieve this. Please figure out this. Besides container, the other way is https://nixos.org/.